ui/src/progress.tsx

```diff
@@ -52,15 +51,21 @@ export interface Progress {
   tooltip?: S
   /** An identifying name for this component. */
   name?: S
+  /** The type of progress bar to be displayed */
+  type?: S
 }
 
 export const
   XProgress = ({ model }: { model: Progress }) => {
     const
-      { label, caption = 'Please wait...', value, name } = model
+      {type, label, caption = 'Please wait...', value, name } = model
     return (
       <div data-test={name}>
-        <Fluent.ProgressIndicator label={label} description={caption} percentComplete={value} />
+        {type === 'spinner' ?
+        <div>
+        <Fluent.Spinner label={label} size={Fluent.SpinnerSize.medium} labelPosition='top'/>
+        </div>
```

Comment:
Seems right i will start working on this

Reply:
need help on make generate step
how to resolve these errors @mturoci 
```
Parsing ../../ui/src/progress.tsx
***Error: code generation failed: unsupported member kind on progress.Progress***
make[1]: *** [Makefile:11: run] Error 1
make[1]: Leaving directory '/home/crash/dev/wave/tools/wavegen'
make: *** [Makefile:138: generate] Error 2
```

Reply:
Hm.. seems like Typescript cannot handle union type with a single value only. Try changing `type?: 'spinner'` to `type?: ' bar' | 'spinner'`. Default should be bar (do not forget to update the docstring). Another possible solution is to use a simple boolean `compact` or `spinner`, but this would lock us out in case we will want to add 3rd type in future. cc @lo5 for discussion.